server/app/src/main/java/com/ams/app/mapper/AttendrecordMapper.java

```diff
@@ -0,0 +1,32 @@
+package com.ams.app.mapper;
+import java.util.List;
+import org.mapstruct.InheritInverseConfiguration;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+import org.springframework.stereotype.Component;
+import com.ams.app.dto.Attendrecorddto;
+import com.ams.app.model.Attendrecord;
+
+@Component
+@Mapper(componentModel = "spring")
+public interface AttendrecordMapper {	
+	AttendrecordMapper INSTANCE = Mappers.getMapper(AttendrecordMapper.class);
+
+    @Mapping(source="faculty.name", target="facultyname")
+    @Mapping(source="faculty.id", target="facultyid")
+    @Mapping(source="subject.name", target="subjectcode")
+    @Mapping(source="subject.name", target="subjectname")
+    @Mapping(source="subject.id", target="subjectid")
```

Comment:
Why do we need so many mappings like this? If needed, you can add FacultyDTO and subjectDTO inside AttendRecordDTO

Reply:
Added Faculty DTO and subject DTO to it